### PR TITLE
Migrate to new Reporter.

### PR DESCRIPTION
**Pull request by dipak-pawar**
#### Short description of what this resolves:
Adds new Reporter integration

#### Changes proposed in this pull request:

- Renamed recorder to reporter from docker.
- Changed docker, k8s reporter to use new reporter.
- Fixed tests with new reporter

**Fixes**: #623

### CODE REVIEW

File: docker/reporter/src/main/java/org/arquillian/cube/docker/impl/client/reporter/DockerLogSection.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+public class DockerLogSection extends SectionEvent<DockerLogSection, BasicReport, TestSuiteSection> {
+
+    public DockerLogSection() {
+        super("DockerLog");
+    }
+
+    @Override
+    public TestSuiteSection getParentSectionThisSectionBelongsTo() {
+        return new TestSuiteSection();
```
Comment (MatousJobanek) [inline docker/reporter/src/main/java/org/arquillian/cube/docker/impl/client/reporter/DockerLogSection.java]: Please, take into account that we could introduce support for multiple test suites in the future. So, make it possible to add the test suite section id - eg. in the same way as it is done in other sections.

File: kubernetes/reporter/src/main/java/org/arquillian/cube/kubernetes/reporter/KubernetesSection.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+public class KubernetesSection extends SectionEvent<KubernetesSection, BasicReport, TestSuiteSection>{
+
+    public KubernetesSection() {
+        super("k8s");
+    }
+
+    @Override
+    public TestSuiteSection getParentSectionThisSectionBelongsTo() {
+        return new TestSuiteSection() ;
```
Comment (MatousJobanek) [inline kubernetes/reporter/src/main/java/org/arquillian/cube/kubernetes/reporter/KubernetesSection.java]: Same as I've commented in DockerLogSection - missing possibility to specify test suite id

File: docker/reporter/pom.xml [hunk 3]
```diff
@@ -15,8 +16,13 @@
```
Comment (MatousJobanek) [inline docker/reporter/pom.xml]: the artifactId of this project still contains `recorder` - change it to `arquillian-cube-docker-reporter`

File: pom.xml [hunk 4]
```diff
@@ -183,6 +184,17 @@
             </dependency>
```
Comment (MatousJobanek) [inline pom.xml]: This still contains the `arquillian-recorder-reporter-*` dependencies - technically, they shouldn't be necessary.

File: kubernetes/reporter/pom.xml [hunk 5]
```diff
@@ -29,6 +24,23 @@
       <groupId>org.arquillian.extension</groupId>
       <artifactId>arquillian-recorder-reporter-api</artifactId>
     </dependency>
```
Comment (MatousJobanek) [inline kubernetes/reporter/pom.xml]: remove `arquillian-recorder-reporter-api` dependency

File: docker/reporter/src/main/java/org/arquillian/cube/docker/impl/client/reporter/DockerLogSection.java (hunk 1 above)
Comment (dipak-pawar) [inline docker/reporter/src/main/java/org/arquillian/cube/docker/impl/client/reporter/DockerLogSection.java]: How would this TestSuiteSection will work if we run multiple test suites in parallel if it has same id. @MatousJobanek

File: docker/reporter/pom.xml (hunk 3 above)
Comment (dipak-pawar) [inline docker/reporter/pom.xml]: :+1:

File: kubernetes/reporter/src/main/java/org/arquillian/cube/kubernetes/reporter/TakeKubernetesResourcesInformation.java [hunk 6]
```diff
@@ -70,19 +52,17 @@ public void reportSessionStatus(@Observes AfterStart afterStart, KubernetesClien
...
-            addEntry(groupEntry, getKeyValueEntry(MASTER_URL, String.valueOf(kubernetesClient.getMasterUrl())));
+            final ReportBuilder reportBuilder = Reporter.createReport(SESSION_STATUS)
+                    .addKeyValueEntry(NAMESPACE, namespace)
+                    .addKeyValueEntry(MASTER_URL, String.valueOf(kubernetesClient.getMasterUrl()));
 
-            addEntry(groupEntry, getTableForReplicationControllers(kubernetesClient, namespace));
-            addEntry(groupEntry, getTableForPods(kubernetesClient, namespace));
-            addEntry(groupEntry, getTableForServices(kubernetesClient, namespace));
+            Reporter.createReport()
+                    .addReport(reportBuilder)
```
Comment (dipak-pawar) [inline kubernetes/reporter/src/main/java/org/arquillian/cube/kubernetes/reporter/TakeKubernetesResourcesInformation.java]: :+1:

File: pom.xml (hunk 4 above)
Comment (dipak-pawar) [inline pom.xml]: removed

**Comment by dipak-pawar** [comment]
@MatousJobanek 
1. I think let's wait for description.
2. We have ./docker/ftest-restassured/pom.xml contains arquillian-reporter-depchain dependency as we have documentation abou restassured reporter. docs/reporter.adoc

**Pull request merged**